### PR TITLE
use recommended setting for use_reentrant w gradient checkpointing

### PR DESCRIPTION
**Pull request by winglian**
fixes
```
packages/torch/utils/checkpoint.py:429: UserWarning: torch.utils.checkpoint: please pass in use_reentrant=True or use_reentrant=False explicitly. The default value of use_reentrant will be updated to be False in the future. To maintain current behavior, pass use_reentrant=True. It is recommended that you use use_reentrant=False. Refer to docs for more details on the differences between the two variants. you may need to update to use_reentrant = False as default when grad check
```

### CODE REVIEW

**Review by hamelsmu — APPROVED** [review]

File: src/axolotl/core/trainer_builder.py [hunk 1]
```diff
@@ -566,6 +566,14 @@ def build(self, total_num_steps):
             training_arguments_kwargs[
                 "gradient_checkpointing"
             ] = self.cfg.gradient_checkpointing
+            if self.cfg.gradient_checkpointing_kwargs:
```
Comment (NanoCode012) [inline src/axolotl/core/trainer_builder.py:569]: Should add this kwargs to Readme
